### PR TITLE
use try-with-resources instead of IOUtils (that is deprecated)

### PR DESCRIPTION
**Pull request by KroArtem**
Update maven-compiler-plugin and set encoding to silence compilation warnings

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,6 +25,7 @@
     </scm>
 
     <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (valfirst) [inline pom.xml:28]: could you please move this change to a separate commit?

File: pom.xml [hunk 2]
```diff
@@ -92,7 +93,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.1</version>
+                <version>3.8.1</version>
```
Comment (valfirst) [inline pom.xml:96]: could you please move this change to one more separate commit?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (KroArtem) [reply pom.xml:96]: You want two more PRs, right?

Comment (valfirst) [reply pom.xml:96]: it doesn't matter: it could 2 more commits in this PR or 2 more PRs
the reason is to keep the changes atomic

File: pom.xml (hunk 1 above)
Comment (KroArtem) [reply pom.xml:28]: done

File: pom.xml (hunk 2 above)
Comment (KroArtem) [reply pom.xml:96]: done

**Review by valfirst — APPROVED** [review]

**Review by pazone — APPROVED** [review]

**Comment by pazone** [comment]
Thanks!

**Comment by KroArtem** [comment]
@pazone , @valfirst , maybe not a good place to ask, anyway, are there any plans for a new version? 1.5.4 was in 2017

**Comment by pazone** [comment]
@KroArtem Will crate a new release workflow on this week.
